applications/mne_analyze/plugins/forwardsolution/forwardsolution.cpp

```diff
@@ -0,0 +1,392 @@
+#include <fwd/computeFwd/compute_fwd_settings.h>
+
+#include <fs/annotationset.h>
+
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QDebug>
+#include <QDockWidget>
+#include <QVBoxLayout>
+
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace FORWARDSOLUTIONPLUGIN;
+using namespace ANSHAREDLIB;
+
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+ForwardSolution::ForwardSolution()
+: m_pCommu(Q_NULLPTR)
+, m_pFwdSettings(new FWDLIB::ComputeFwdSettings)
+, m_bBusy(false)
+, m_bDoRecomputation(false)
+, m_bDoClustering(true)
+, m_bDoFwdComputation(false)
+{
+    // set init values
+    m_pFwdSettings->solname = QCoreApplication::applicationDirPath() + "/MNE-sample-data/your-solution-fwd.fif";
+    m_pFwdSettings->mriname = QCoreApplication::applicationDirPath() + "/MNE-sample-data/MEG/sample/all-trans.fif";
+    m_pFwdSettings->bemname = QCoreApplication::applicationDirPath() + "/MNE-sample-data/subjects/sample/bem/sample-5120-5120-5120-bem.fif";
+    m_pFwdSettings->srcname = QCoreApplication::applicationDirPath() + "/MNE-sample-data/subjects/sample/bem/sample-oct-6-src.fif";
+    m_pFwdSettings->measname = QCoreApplication::applicationDirPath() + "/MNE-sample-data/MEG/sample/sample_audvis_raw.fif";
+    m_pFwdSettings->transname.clear();
+    m_pFwdSettings->eeg_model_name = "Default";
+    m_pFwdSettings->include_meg = true;
+    m_pFwdSettings->include_eeg = true;
+    m_pFwdSettings->accurate = true;
+    m_pFwdSettings->mindist = 5.0f/1000.0f;
+
+    m_sAtlasDir = QCoreApplication::applicationDirPath() + "/MNE-sample-data/subjects/sample/label";
+
+    m_pAnnotationSet = FSLIB::AnnotationSet::SPtr(new FSLIB::AnnotationSet(m_sAtlasDir+"/lh.aparc.a2009s.annot", m_sAtlasDir+"/rh.aparc.a2009s.annot"));
+}
+
+//=============================================================================================================
+
+ForwardSolution::~ForwardSolution()
+{
+}
+
+//=============================================================================================================
+
+QSharedPointer<AbstractPlugin> ForwardSolution::clone() const
+{
+    QSharedPointer<ForwardSolution> pForwardSolutionClone(new ForwardSolution);
+    return pForwardSolutionClone;
+}
+
+//=============================================================================================================
+
+void ForwardSolution::init()
```

Comment:
Try to avoid two phase initialisation -> https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nr5-dont-use-two-phase-initialization

I know I probably introduced this a long time ago ;)